File: Data Structures & Algorithms/validate-parentheses/submission-12.py
class Solution:
    def isValid(self, s: str) -> bool:
        map={"}":"{","]":"[",")":"("}
        stack = []
        if len(s) <2:
            return False
        for i in s:
            if i == "[" or i == "{" or i =="(":
                stack.append(i)
            else:
                if stack and map[i] != stack.pop():
                    return False
        if stack :
            return False
        else:
            return True
                
        

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-12.java
class Solution {
    public int evalRPN(String[] tokens) {
        Stack <Integer>stack = new Stack<>();
        for(String i :tokens){
            if (i.equals("+")){
            int a = stack.pop();
            int b = stack.pop();
            stack.push(a+b);
            }
            else if(i.equals("*")){
                int a = stack.pop();
                int b = stack.pop();
                stack.push(a*b);
            }
            else if(i.equals("-")){
                int a = stack.pop();
                int b = stack.pop();
                stack.push(a-b);
            }
            else if (i.equals("/")){
                int a = stack.pop();
                int b = stack.pop(); 
                int c = (int)b/a;
                stack.push(c);
            }
            else{
                int c = Integer.parseInt(i);
                stack.push(c);
            }
        }
        int c =stack.pop();
        return c;
        
    }
}
